okhttp-clients/src/main/java/com/palantir/conjure/java/okhttp/RemotingOkHttpCall.java

```diff
@@ -175,24 +177,31 @@ public void onFailure(Call call, IOException exception) {
                 // Fail call if backoffs are exhausted or if no retry URL can be determined.
                 Optional<Duration> backoff = backoffStrategy.nextBackoff();
                 if (!backoff.isPresent()) {
-                    callback.onFailure(call, new IOException("Failed to complete the request due to an "
-                            + "IOException", exception));
+                    callback.onFailure(call, new SafeIoException(
+                            "Failed to complete the request due to an IOException",
+                            exception,
+                            UnsafeArg.of("requestUrl", call.request().url().toString())));
                     return;
                 }
+
                 Optional<HttpUrl> redirectTo = urls.redirectToNext(request().url());
                 if (!redirectTo.isPresent()) {
-                    callback.onFailure(call, new IOException("Failed to determine valid failover URL for '"
-                            + request().url() + "' and base URLs " + urls.getBaseUrls()));
+                    callback.onFailure(call, new SafeIoException(
+                            "Failed to determine valid failover URL",
+                            exception,
+                            UnsafeArg.of("requestUrl", call.request().url().toString()),
+                            UnsafeArg.of("baseUrls", urls.getBaseUrls())));
                     return;
                 }
 
+                log.info("Retrying call after failure",
+                        SafeArg.of("backoffMillis", backoff.get().toMillis()),
```

Comment:
we should log the `redirectTo.get()` url as well, so it's obvious what server it will try nextt